x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/indexlifecycle/UnfollowFollowIndexStep.java

```diff
@@ -25,7 +31,19 @@ void innerPerformAction(String followerIndex, Listener listener) {
                 assert r.isAcknowledged() : "unfollow response is not acknowledged";
                 listener.onResponse(true);
             },
-            listener::onFailure
+            exception -> {
+                if (exception instanceof ElasticsearchException
+                        && ((ElasticsearchException) exception).getMetadata("es.failed_to_remove_retention_leases") != null) {
+                    List<String> leasesNotRemoved = ((ElasticsearchException) exception)
+                        .getMetadata("es.failed_to_remove_retention_leases");
+                    logger.debug("Failed to remove leader retention lease(s) {} while unfollowing index [{}], " +
```

Comment:
I wonder if this should be warn?

Reply:
I wouldn't be opposed to making this `warn`, I only made it `debug` because it looks like [TransportUnfollowAction already emits a warning](https://github.com/elastic/elasticsearch/pull/39088/files#diff-93f9315523c24e85103be733259dad9fR129) in this case.

Reply:
Both unfollow execution and ilm policy execution happen on the elected master node, so I think logging twice a warning is not needed here.

Reply:
Okay, that is fair.